incubator/hnc/internal/validators/hierarchy.go

```diff
@@ -176,6 +176,11 @@ func (v *Hierarchy) checkNS(ns *forest.Namespace) admission.Response {
 
 // checkParent validates if the parent is legal based on the current in-memory state of the forest.
 func (v *Hierarchy) checkParent(ns, curParent, newParent *forest.Namespace) admission.Response {
+	if ns.IsExternal() && newParent != nil {
+		msg := fmt.Sprintf("Namespace %q is an external namespace that cannot have parent. Please unset the parent if it exists.", ns.Name())
```

Comment:
Maybe replace "is an external namespace" to "is managed by %q"?

Reply:
I'd also just leave out the "Please unset the parent if it exists" part, it's fine to just say "it can't have a parent."